src/main/java/bio/terra/janitor/service/iam/AuthenticatedUserRequest.java

```diff
@@ -0,0 +1,67 @@
+package bio.terra.janitor.service.iam;
+
+import bio.terra.janitor.common.exception.ApiException;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import java.util.Optional;
+import java.util.UUID;
+
+public class AuthenticatedUserRequest {
+  private String email;
+  private String subjectId;
+  private Optional<String> token;
+  private UUID reqId;
+
+  public AuthenticatedUserRequest() {
+    this.reqId = UUID.randomUUID();
+  }
+
+  public AuthenticatedUserRequest(String email, String subjectId, Optional<String> token) {
```

Comment:
Taking an optional as constructor / method param is a bit questionable (the most convincing argument I've seen is something like https://stackoverflow.com/a/39005452). I also can't find any usage of the token field in this PR. Following go/yagni principle, maybe scrap the token field altogether for now?

Reply:
Right. Done. Sorry for so many confusions in this classes. When I copied the whole thing from WM, I didn't pay too much attention to refactor this class. Seems WM and TDR has more complicated use cases as they use SAM. Deleted the token for now.